code/handlers/trackservers.q

```diff
@@ -177,11 +173,16 @@ autodiscovery:{if[DISCOVERYRETRY>0; .servers.retrydiscovery[]]}
 
 // Attempt to make a connection for specified row ids
 retryrows:{[rows]
+    //function a checks if the handle passed is empty and also invokes checknontorqattr function 
+    //which checks if .proc.getattributes is defined on the nontorqprocess and executes it 
+    //only if it is defined
+    a:{$[(first not null x)&@[first x;"`getattributes in key`.proc";0b];:@[first x;(`.proc.getattributes;`);()!()];()!()]};
```

Comment:
Why do you have the "first"s in here?  Are there cases where x is passed in as a list? 

This current version is making two calls to the remote process - can you just make it one call?  So a lambda which checks if getattributes exists and then calls it if it does?